tt_metal/llrt/hal.hpp

```diff
@@ -30,6 +30,23 @@ enum class HalProgrammableCoreType {
     COUNT      = 3
 };
 
+enum class HalProcessorClassType {
+    DM0     = 0,
+    DM1     = 1,
+    COMPUTE = 2,
```

Comment:
Isn't this DM and COMPUTE?  Then there are 1 or 2 instances of DM depending on core type?